aviary/subsystems/propulsion/propulsion_mission.py

```diff
@@ -94,28 +94,37 @@ def setup(self):
                     self.promotes(engine.name, inputs=[Dynamic.Mission.HYBRID_THROTTLE])
 
         # TODO might be able to avoid hardcoding using propulsion Enums
-        # mux component to vectorize individual outputs into 2d arrays
+        # mux component to vectorize individual engine outputs into 2d arrays
         perf_mux = om.MuxComp(vec_size=num_engine_type)
         # add each engine data variable to mux component
-        perf_mux.add_var(Dynamic.Mission.THRUST, shape=(nn,), axis=1, units='lbf')
-        perf_mux.add_var(Dynamic.Mission.THRUST_MAX, shape=(nn,), axis=1, units='lbf')
         perf_mux.add_var(
-            Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE, shape=(nn,), axis=1, units='lbm/h'
+            Dynamic.Mission.THRUST, val=0, shape=(nn,), axis=1, units='lbf'
         )
         perf_mux.add_var(
-            Dynamic.Mission.ELECTRIC_POWER_IN, shape=(nn,), axis=1, units='kW'
+            Dynamic.Mission.THRUST_MAX, val=0, shape=(nn,), axis=1, units='lbf'
         )
-        perf_mux.add_var(Dynamic.Mission.NOX_RATE, shape=(nn,), axis=1, units='lb/h')
         perf_mux.add_var(
-            Dynamic.Mission.TEMPERATURE_T4, shape=(nn,), axis=1, units='degR'
+            Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+            val=0,
+            shape=(nn,),
+            axis=1,
+            units='lbm/h',
+        )
+        perf_mux.add_var(
+            Dynamic.Mission.ELECTRIC_POWER_IN, val=0, shape=(nn,), axis=1, units='kW'
+        )
+        perf_mux.add_var(
+            Dynamic.Mission.NOX_RATE, val=0, shape=(nn,), axis=1, units='lb/h'
+        )
+        perf_mux.add_var(
+            Dynamic.Mission.TEMPERATURE_T4, val=0, shape=(nn,), axis=1, units='degR'
+        )
+        perf_mux.add_var(
+            Dynamic.Mission.SHAFT_POWER, val=0, shape=(nn,), axis=1, units='hp'
+        )
+        perf_mux.add_var(
+            Dynamic.Mission.SHAFT_POWER_MAX, val=0, shape=(nn,), axis=1, units='hp'
```

Comment:
should these be added based on the variables that are actually present in the engines, or are these all needed downstream of the mux (just not required from the engine model)

Reply:
I chose to keep all the variables present post-muxing so there won't be any problems with anything downstream asking for these values (like the fuel summation comp for an all-electric aircraft). Thrust is also definitely used by the EOMs too. For the most part I don't think most of these have to be always present, but it also doesn't hurt to just have them available and defaulted to zero just in case someone eventually does want them.